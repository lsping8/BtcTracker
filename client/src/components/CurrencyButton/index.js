import React, { Component } from 'react';
import { offQuery } from '../../utils/FirebaseService'

class CurrencyButton extends Component {

  handleClick = (newCurrency) => {
    const { changeCurrency,queryData,currency } = this.props
    console.log('newCurrency',newCurrency);
    offQuery(currency)
    changeCurrency(newCurrency)
    queryData(newCurrency)
  }

  render() {
    const { currency } = this.props
    return (
      <div className="button-grid">
        <div
          className={currency === 'USD' ? "box-select" : 'box'}
          style={{padding: 20,cursor:'pointer'}}
          onClick={() => this.handleClick('USD')}
        >
          USD
        </div>
        <div
          className={currency === 'SGD' ? "box-select" : 'box'}
          style={{padding: 20,cursor:'pointer'}}
          onClick={() => this.handleClick('SGD')}
        >
          SGD
        </div>
      </div>
    );
  }
}

export default CurrencyButton;
