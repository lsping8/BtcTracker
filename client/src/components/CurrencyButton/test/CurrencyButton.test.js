import React from 'react'
import ReactDOM from 'react-dom'
import { shallow,configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import CurrencyButton from '../index'

configure({ adapter: new Adapter() });

describe('CurrencyButton', () => {
  const props = { currency: 'USD' }
  const currencyButton = shallow(<CurrencyButton {...props}/>)
  it('renders without crashing', () => {
    expect(currencyButton).toMatchSnapshot()
  })

  it('current selected currency', () => {
    expect(currencyButton.find('.box-select').text()).toEqual('USD')
  })

  it('current unselect currency', () => {
    expect(currencyButton.find('.box').text()).toEqual('SGD')
  })
})
