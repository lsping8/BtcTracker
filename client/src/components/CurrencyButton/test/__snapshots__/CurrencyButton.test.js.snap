// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyButton renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyButton
    currency="USD"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-select"
          onClick={[Function]}
          style={
                    Object {
                              "cursor": "pointer",
                              "padding": 20,
                            }
          }
>
          USD
</div>,
        <div
          className="box"
          onClick={[Function]}
          style={
                    Object {
                              "cursor": "pointer",
                              "padding": 20,
                            }
          }
>
          SGD
</div>,
      ],
      "className": "button-grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "USD",
          "className": "box-select",
          "onClick": [Function],
          "style": Object {
            "cursor": "pointer",
            "padding": 20,
          },
        },
        "ref": null,
        "rendered": "USD",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SGD",
          "className": "box",
          "onClick": [Function],
          "style": Object {
            "cursor": "pointer",
            "padding": 20,
          },
        },
        "ref": null,
        "rendered": "SGD",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-select"
            onClick={[Function]}
            style={
                        Object {
                                    "cursor": "pointer",
                                    "padding": 20,
                                  }
            }
>
            USD
</div>,
          <div
            className="box"
            onClick={[Function]}
            style={
                        Object {
                                    "cursor": "pointer",
                                    "padding": 20,
                                  }
            }
>
            SGD
</div>,
        ],
        "className": "button-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "USD",
            "className": "box-select",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
              "padding": 20,
            },
          },
          "ref": null,
          "rendered": "USD",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SGD",
            "className": "box",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
              "padding": 20,
            },
          },
          "ref": null,
          "rendered": "SGD",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
